### PR TITLE
[Concerts] Fix saving Extra Fanarts

### PR DESCRIPTION
**Pull request by alexls74**

### CODE REVIEW

File: concerts/ConcertController.cpp [hunk 1]
```diff
@@ -218,6 +218,9 @@ void ConcertController::onDownloadFinished(DownloadManagerElement elem)
             Manager::instance()->mediaCenterInterface()->imageFileName(m_concert, elem.imageType));
         if (elem.imageType == ImageType::ConcertBackdrop) {
             Helper::instance()->resizeBackdrop(elem.data);
+        } else if (!elem.data.isEmpty() && elem.imageType == ImageType::ConcertExtraFanart) {
```
Comment (bugwelle) [inline concerts/ConcertController.cpp]: `!elem.data.isEmpty()` is unnecessary (see line 216).

File: concerts/ConcertController.cpp [hunk 2]
```diff
@@ -218,6 +218,9 @@ void ConcertController::onDownloadFinished(DownloadManagerElement elem)
...
+            Helper::instance()->resizeBackdrop(elem.data);
+            m_concert->addExtraFanart(elem.data);
```
Comment (bugwelle) [inline concerts/ConcertController.cpp]: Could you please elaborate why you call `m_concert->addExtraFanart(...);` as well as `setImage()` in line 221?
I had a look at `MovieController.cpp` (line 371) and it seems `setImage` shouldn't be called when you add extra fanart. I would change this to:

```cpp
if (!elem.data.isEmpty()) {
    if (elem.imageType == ImageType::ConcertExtraFanart) {
        Helper::instance()->resizeBackdrop(elem.data);
        m_concert->addExtraFanart(elem.data);
    } else {
        ImageCache::instance()->invalidateImages(
                Manager::instance()->mediaCenterInterface()->imageFileName(m_concert, elem.imageType));
        if (elem.imageType == ImageType::ConcertBackdrop) {
            Helper::instance()->resizeBackdrop(elem.data);
        }
        m_concert->setImage(elem.imageType, elem.data);
    }
}
```

**Comment by alexls74** [comment]
> Was extra fanart just not loaded or not saved on disk?

Downloaded but not saved

> I would change this to: ...

Done

**Comment by bugwelle** [comment]
Thanks! Spotting those bugs is a lot of work. Maybe just a few more months and we're bug free 😄

**Pull request merged**